### ISSUE TITLE
False positive on custom elements using formAssociated

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#4750]
This should not fail `aria-input-field-name`.

```html
<script>
  class FormInput extends HTMLElement {
    static formAssociated = true
    #input
    constructor() {
      super()
      const root = this.attachShadow({ mode: 'open' })
      root.innerHTML = ` <input type="text" aria-hidden="true"> `
    }
  }
  customElements.define('form-input', FormInput)
</script>
<!-- implicit label -->
<label> Hello <form-input role="textbox"></form-input> </label>
<!-- explicit label -->
<label for="world"> World </label>
<form-input role="textbox" id="world"></form-input>
```

On the surface this doesn't look too difficult to figure out. Check the custom element's `.constructor.formAssociated` prop. The difficulty is that this property is only available when axe runs in the same javascript world as the page. Playwright, Puppeteer, Chromium extensions all run inside a different javascript world. There are ways to work around that by appending a script to the page. That could trigger a custom event that can pass information back. I'm not sure that's desirable.

An alternative to consider is to raise this as a Chromium bug. I haven't checked what other browsers do with this. It's kind of for a DOM property not to be available in all worlds. Then again, this really doesn't function like normal properties do. The browser reads this property once when `customElement.define` is called. Changing it afterwards does nothing. I.e. you cannot turn off formAssociation.
